src/python/gudhi/representations/vector_methods.py

```diff
@@ -806,32 +807,49 @@ def fit(self, X, y=None, sample_weight=None):
         Returns:
             self
         """
-        if not hasattr(self.quantiser, 'fit'):
-            raise TypeError("quantiser %s has no `fit` attribute." % (self.quantiser))
-
-        # In fitting we remove infinite death time points so that every center is finite
-        X = [dgm[~np.isinf(dgm).any(axis=1), :] for dgm in X]
+        n_clusters = self.quantiser.n_clusters
 
+        if not len(X):
+            raise Exception("Cannot fit Atol on empty target.")
+        measures_concat = np.concatenate(X)
         if sample_weight is None:
             sample_weight = [self.get_weighting_method()(measure) for measure in X]
-
-        measures_concat = np.concatenate(X)
         weights_concat = np.concatenate(sample_weight)
 
-        self.quantiser.fit(X=measures_concat, sample_weight=weights_concat)
+        # In fitting we remove infinite birth/death time points so that every center is finite. We do not care about duplicates.
+        filtered_measures_concat = measures_concat[~np.isinf(measures_concat).any(axis=1), :] if len(measures_concat) else measures_concat
+        filtered_weights_concat = weights_concat[~np.isinf(measures_concat).any(axis=1)] if len(measures_concat) else weights_concat
+        n_points = len(filtered_measures_concat)
+        if not n_points:
+            raise Exception("Cannot fit Atol on measure with infinite components only.")
 
+        if n_points < n_clusters:
+            # If not enough points to fit (including 0), we will arbitrarily put centers as [-np.inf]^measure_dim at the end.
+            print(f"[Atol] had {n_points} points to fit {n_clusters} clusters, adding meaningless cluster centers.")
```

Comment:
We could also use warnings here, no ?
```suggestion
            warnings.warn(f"[Atol] had {n_points} points to fit {n_clusters} clusters, adding meaningless cluster centers.", RuntimeWarning)
```

Also requires at the beginning of file:
```python
import warnings
```

Reply:
A drawback of warnings.warn is that by default, IIRC, it warns only once. That's good for something like a deprecation warning, or a missing dependency. But for this kind of runtime warning, I think we want it to appear "always" by default (unless the user explicitly overrides it of course). It is probably possible.

Reply:
Yes it is up to the end user to activate it. But `print` may be not strong enough in this case.
But ok, keep it like that, but we have to think on how we want to handle these cases.

Reply:
If we want the warning to appear multiple times, one possibility is
```python
    with warnings.catch_warnings():
        warnings.simplefilter("always")
        warnings.warn("foo", RuntimeWarning)
```
Another one is to ensure that the message is different every time (with a global counter?).
A last one is to use print (preferably to sys.stderr).

I didn't see a convenient way to say: "if you would print the warning if this was the first time, print it anyway, but if warnings have been disabled, don't", the closest I came is replacing the second line above with `warnings.simplefilter("always", append=True)`, i.e. "if there is no explicit filter, default to 'always' for this warning", which doesn't seem too bad.

Reply:
I first implemented the warnings suggestion then rolled back on it because I couldn't clearly see the benefits over `print`: it doesn't seem to me (might be wrong) that the situation where we add infinite "bogus" centers is particularly dangerous / bad for the user that it needs special attention.

Reply:
I wouldn't say this print qualifies as a warning or error or abnormal output?, I would say it is additional information on the nature of your fit data `X` with respect to your expectations on the size of the vectorization. Maybe I misunderstood, but it would seem weird to me to give _this_ print a specific treatment compared to others on the file. We could also remove it altogether.

Reply:
Let me rephrase. My comment applied to *all* the `print` in this file, not just this one (I have been lazy on enforcing this). Which also means that it won't be a requirement for this PR (we can change them all later in another PR).

The idea is:
* I write a script.py that computes some Atol transform, and prints the output (maybe "1.5 2.3\n4.0 0.").
* I run `python script.py > vect.txt` to redirect the output and store it in a file.
* I now run another program that reads vect.txt (say with numpy.loadtxt)

If I have a string "adding meaningless cluster centers" in the middle of vect.txt, the program reading it will get confused and fail. If we use `print("bla", file=sys.stderr)` in Atol, then `python script.py > vect.txt` will print "bla" on the console and not in the file (it is a separate stream, which could be redirected separately with `2> log.txt`), and the file remains unpolluted and usable by further programs.
`stderr` is for any kind of notice, warning, etc, not just errors.

Is the principle of stdout (the default for `print`) vs stderr clearer this way?

Reply:
Yes I understand the idea of stderr / stdout / .... To that end the logging module is much cleaner if we want to go that direction.